robot/Cumulus/resources/RecurringDonationsPageObject.py

```diff
@@ -88,17 +96,23 @@ def edit_recurring_donation_status(self,**kwargs):
 
     @capture_screenshot_on_error
     def _populate_edit_status_values(self, **kwargs):
-        """Pass the status and reason for the status as key, value pairs to populate the edit form"""
+        """Takes the key value pairs to edit and makes changes accordingly"""
         for key, value in kwargs.items():
-            locator = npsp_lex_locators["erd"]["modal_dropdown_selector"].format(key)
-            selection_value = npsp_lex_locators["erd"]["modal_selection_value"].format(value)
-            if self.npsp.check_if_element_exists(locator):
-                self.selenium.set_focus_to_element(locator)
-                self.selenium.wait_until_element_is_visible(locator)
-                self.selenium.scroll_element_into_view(locator)
-                self.salesforce._jsclick(locator)
-                self.selenium.wait_until_element_is_visible(selection_value)
-                self.selenium.click_element(selection_value)
+            if key in ("Amount", "Number of Planned Installments", "Every"):
+                locator = npsp_lex_locators["erd"]["modal_input_field"].format(key)
+                if self.npsp.check_if_element_exists(locator):
+                    self.selenium.set_focus_to_element(locator)
+                    self.salesforce._populate_field(locator, value)
```

Comment:
If the locator doesn't exists, the keyword does nothing ?
How about adding a log message or an exception at the end to catch such a scenario ?

Reply:
Added else condition and log